### DISCUSSION TITLE
Unable to connect to Klipper firmware via USB

### DISCUSSION THREAD

**Discussion by Thibaud-P** [bassamanator/Sovol-SV06-firmware#79] [Q&A] [answered]
Hi,

I'm sorry to have to ask for help, but I tried everything I could...

I don't have a RPi or alternative right now, I'm trying to use Klipper on Windows via WSL, so far so good, I managed to get everything right (install Klipper, Moonraker, Mainsail, change user in config files, etc.).
I flashed the firmware in the repo under misc.
But Klippy can't connect to the MCU.

I tried compiling different firmware, changing UART1 to UART2 or USB, or using internal clock, each time with a new name as the only file on the sd card.
Nothing better.

I flashed the original firmware (same sd card) and managed to connect without problems (same WSL setup but with Octoprint, same cable).
Also, the connection works with a local non WSL windows octoprint install, pronterface, etc...
So I'm pretty sure it is a firmware problem, or a flashing firmware problem (but I manage to flash the original firmware back with the sd card I use).

My Sovol SV06 board says version 1.3, it seems like other people managed to get it working without problems ?
I followed the instuctions on https://github.com/bassamanator/Sovol-SV06-firmware/discussions/78 to compile the firmware for my tests.
Any idea on what to try now ?

I need some way to confirm that the firmware flashed correctly, and then something to try the serial communication while under the Klipper firmware.

Thanks for your help,

Thibaud

**Comment by bassamanator** [comment, marked as answer]
The only way I know that a Klipper flash succeeded, is by making sure everything is connected, then simply doing a `firmware_restart` in the klipper console. If everything is working, you will see your LCD screen light up and show you some stats (temp, fans, etc.).

I'm curious, how did you find the `mcu` path to use in `printer.cfg`? `Adjust Configuration with MCU Path` section.

**Reply by Thibaud-P** [reply]
Wow I tried the exact same steps from the beginning to generate clean log for further investigations, and it just worked Oo.
With the firmware in your repo. I've got connection on mainsail, and the temp etc on screen.

Thanks for your help, it seems like you sent the positive energy missing to make it work !
I can't explain it.

Great job gathering all those informations here, everything :)
And sorry for the inconvenience !

Now time to test Klipper

**Reply by bassamanator** [reply]
My pleasure. 😁

Please close this discussion when ready. And feel free to re-open.
